rosidl_runtime_c/test/test_string_functions.cpp

```diff
@@ -116,6 +116,32 @@ TEST(string_functions, resize_assignn) {
   rosidl_runtime_c__String__fini(&s);
 }
 
+TEST(string_functions, equality_comparison) {
+  rosidl_runtime_c__String empty_string, foo_string, bar_string;
+  EXPECT_TRUE(rosidl_runtime_c__String__init(&empty_string));
```

Comment:
These could be asserts to break early if initialisation fails.

Reply:
My understanding is that no tests in here do that to ensure `rosidl_runtime_c__String__fini` gets invoked. I could bring in `rcpputils::scope_exit` though. Thoughts?